app/flatpak-builtins-build.c

```diff
@@ -75,6 +75,94 @@ child_setup (gpointer user_data)
     fcntl (g_array_index (fd_array, int, i), F_SETFD, 0);
 }
 
+static gboolean
+find_matching_extension_group_in_metakey (GKeyFile    *metakey,
+                                          const char  *id,
+                                          const char  *specified_tag,
+                                          char       **out_extension_group,
+                                          GError     **error)
+{
+  g_auto(GStrv) groups = NULL;
+  g_autofree char *extension_prefix = NULL;
+  const char *untagged_extension_group = NULL;
+  const char *last_seen_group = NULL;
+  guint n_extension_groups = 0;
+  GStrv iter = NULL;
+
+  g_return_val_if_fail (out_extension_group != NULL, FALSE);
+
+  groups =  g_key_file_get_groups (metakey, NULL);
+  extension_prefix = g_strconcat (FLATPAK_METADATA_GROUP_PREFIX_EXTENSION,
+                                  id,
+                                  NULL);
+
+  for (iter = groups; *iter != NULL; ++iter)
+    {
+      const char *group_name = *iter;
+      const char *extension_name = group_name + strlen (FLATPAK_METADATA_GROUP_PREFIX_EXTENSION);
+      g_autofree char *extension_tag = NULL;
+
+      flatpak_parse_extension_with_tag (extension_name,
```

Comment:
You can't looke at extension name before you verified that group_name has the right prefix, it could point past the end of group_name. I recommend to start the for loop with `if (!has_prefix) continue;`

Reply:
Yes, good catch